### ISSUE TITLE
Prefer `event.key` over `event.keyCode`

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#118]
[<img alt="Issuehunt badges" src="https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/118/badge.svg" />](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/118)

For readability. Got the idea when doing https://github.com/wulkano/kap/commit/71be04f7411aa33d0b393c05077c619de2ab07be.

Should trigger when in an `addEventListener` callback.

## Fail

```js
foo.addEventListener('click', event => {
	if (event.keyCode === '27') {
		…
	}
});
```

## Pass

```js
foo.addEventListener('click', event => {
	if (event.key === 'Escape') {
		…
	}
});
```

More info: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

---

For the first implementation, we can just naively look for the same variable name as defined as the parameter. Usually `event` or `e`.

Later on, we could add code path support and support having the function outside the `addEventListener` call:

```js
const handler = event => {
	if (event.key === 'Escape') {
		…
	}
};

…

foo.addEventListener('click', handler);
```

---

@kevva @SamVerschueren Thoughts?

> There is a $80.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/118).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

**Comment by SamVerschueren** [comment]
Much clearer to reason about! Would be nice to have!

**Comment by IssuehuntBot** [comment]
@issuehuntfest has funded $80.00 to this issue. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/118)

**Comment by futpib** [comment]
Do we want:
* to report any `event.keyCode` reference or only comparisons like `event.keyCode === 27`?
* a fixer that turns `event.keyCode === 27` into `event.key === 'Escape'`?
* to also report [`event.charCode`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/charCode)?
* to also report [`event.which`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which)?
* to have this rule enabled by default?

**Comment by sindresorhus** [comment]
> to report any event.keyCode reference or only comparisons like event.keyCode === 27?

I would say any usage. I can't think of a situation where it would make more sense to use `keyCode` over `key`. Can you?

> a fixer that turns event.keyCode === 27 into event.key === 'Escape'?

👍 

> to also report event.charCode?

👍 

> to also report event.which?

👍 

> to have this rule enabled by default?

👍

**Comment by maazadeeb** [comment]
@sindresorhus I'd like to work on this. Can I take it up?

**Comment by ankeetmaini** [comment]
Can I take a stab at it?

**Comment by IssueHuntBot** [comment]
@sindresorhus has rewarded $72.00 to @ankeetmaini. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/118)

- :moneybag: Total deposit: $80.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $8.00
